python/lsst/daf/butler/core/dimensions/_records.py

```diff
@@ -175,6 +181,105 @@ def __reduce__(self) -> tuple:
         mapping = {name: getattr(self, name) for name in self.__slots__}
         return (_reconstructDimensionRecord, (self.definition, mapping))
 
+    def to_simple(self, minimal: bool = False) -> Dict[str, Any]:
+        """Convert this class to a simple python type suitable for
+        serialization.
+
+        Parameters
+        ----------
+        minimal : `bool`, optional
+            Use minimal serialization. Has no effect on for this class.
+
+        Returns
+        -------
+        names : `list`
+            The names of the dimensions.
+        """
+        if minimal:
+            # The DataId is sufficient if you are willing to do a deferred
+            # query. This may not be overly useful since to reconstruct
+            # a collection of records will require repeated registry queries.
+            simple = self.dataId.to_simple()
+            # Need some means of indicating this is not a full record
+            simple["element"] = self.definition.name
+            return simple
+
+        mapping = {name: getattr(self, name) for name in self.__slots__}
+        # If the item in mapping supports simplification update it
+        for k, v in mapping.items():
+            try:
+                mapping[k] = v.to_simple(minimal=minimal)
+            except AttributeError:
+                if isinstance(v, lsst.sphgeom.Region):
+                    # Match YAML serialization
+                    mapping[k] = {"cls": f"lsst.sphgeom.{type(v).__name__}",
+                                  "encoded": v.encode().hex()}
```

Comment:
I actually noticed the other day that the sphgeom class is already included in the encoded representation, and you can use `Region.decode` to handle a byte string produced by any derived class.  It's might be too late to be worth switching the YAML representation, but we could get away with just `bytes` for this one, if we wanted to.

Reply:
I noticed too that the YAML form was storing the class and then not using it. It seems like I messed up somewhere as we iterated to the final form. On the other hand the `cls` here does make it obvious that it's a region although using a single key in a dict of `encoded_region` would also do that.

Reply:
Changed it to just use the "encoded_region" key. It can't use bytes because json requires strings.